spec/controllers/bridges/lom/tasks_controller_spec.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Bridges::Lom::TasksController do
+  render_views
+
+  let(:user) { create(:user) }
+  let(:programming_language) { create(:programming_language) }
+  let(:license) { create(:license) }
+  let(:parent_uuid) { create(:task).uuid }
+  let(:public_task) { create(:task, :with_content, user:, parent_uuid:, programming_language:, license:, access_level: :public) }
+  let(:private_task) { create(:task, :with_content, user:, parent_uuid:, programming_language:, license:, access_level: :private) }
+
+  describe 'GET #show' do
+    subject(:get_request) { get :show, params: {id: task.to_param} }
+
+    shared_examples 'validate response LOM' do
+      it 'returns HTTP OK' do
+        expect(response).to have_http_status :ok
+      end
+
+      it 'returns valid LOM xml' do
+        get_request
+        schema = Nokogiri::XML::Schema(File.read(Bridges::Lom::TasksController::OML_SCHEMA_PATH))
+        expect(schema.validate(Nokogiri::XML(response.body))).to be_empty
+      end
+    end
+
+    context 'without being signed in' do
+      let(:task) { public_task }
+
+      it 'returns HTTP forbidden' do
+        get_request
+        expect(response).to have_http_status :forbidden
+      end
```

Comment:
Right, I got this confused because we said we did not want any task to be accessible for users that are not signed in but on the other hand we do allow access to the LOM.

Reply:
Not a problem at all ;)